### PR TITLE
spike-dasm: Add Snitch ISA aware riscv-isa-sim as submodule

### PR DESCRIPTION
**Pull request by GiannaP**
Add Snitch ISA aware `riscv-isa-sim` version as submodule. Only used for `spike-dasm` to make the generated traces more readable.

In the old `snitch` repository it was flattened (without `vendor`) from the [repository](https://github.com/huettern/riscv-isa-sim) of @huettern. The branch was ported to the [pulp-platform repository](https://github.com/pulp-platform/riscv-isa-sim/tree/GiannaP-snitch).

### CODE REVIEW

File: docs/ug/getting_started.md [hunk 1]
```diff
@@ -49,8 +49,18 @@ export CC="gcc-9.2.0"
...
-export PATH=/home/colluca/snitch/bin:$PATH
 export PATH=/usr/scratch/dachstein/colluca/opt/verible/bin:$PATH
+
+# Install spike for spike-dasm (submodule of the repository)
+cd sw/deps/riscv-isa-sim
+autoconf
+mkdir build
+cd build
+../configure
+make
```
Comment (colluca) [inline docs/ug/getting_started.md:60]: Installation instructions provided do not terminate successfully

**Comment by paulsc96** [comment]
Behold my new idea: the *limbo* tag :smiling_face_with_tear: It labels PRs with long-term unresolved discussions. No, jokes aside, please look into this or redraft this PR.

File: docs/ug/getting_started.md (hunk 1 above)
Comment (GiannaP) [reply docs/ug/getting_started.md:66]: To keep track of debugging:
It seems to work with GCC version `8.5.0` and `4.8.5`, but not with `9.2.0`

Running `make` with version `9.2.0` fails with the following error:
```
g++-9.2.0 -MMD -MP  -Wall -Wno-unused -g -O2 -std=c++11 -DPREFIX=\"/usr/local\" -I. -I.. -I../fesvr -I../riscv -I../dummy_rocc -I../softfloat -I../spike_main -x c++-header ../riscv/insn_template.h -o insn_template.h.gch
/lib/../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
collect2: error: ld returned 1 exit status
make: *** [insn_template.h.gch] Error 1
```

**Comment by colluca** [comment]
I added the `spike-dasm` installation to the Docker container.

Last thing missing is a proper flow for installation on IIS machines. Specifically:
- settle on a set of reproducible instructions (e.g. with a specific GCC version)
- add the installation instructions to the CI (either on the `snitch_cluster` or `riscv_isa_sim` repo)
- distribute a pre-built `spike-dasm` binary to IIS machines or provide instructions to install it (either on the `snitch_cluster` repo or point to `riscv_isa_sim` repo)

File: util/container/Dockerfile [hunk 2]
```diff
@@ -119,6 +128,8 @@ RUN apt-get update && apt-get install software-properties-common -y && \
 # Copy artifacts from stage 1.
 COPY --from=builder /root/.cargo/bin/bender bin/
 COPY --from=builder /root/.cargo/bin/banshee bin/
+COPY --from=builder /root/.cargo/bin/banshee bin/
```
Comment (Raragyay) [inline util/container/Dockerfile:131]:
```suggestion
```

**Comment by colluca** [comment]
I added releases to the [riscv-isa-sim repo](https://github.com/pulp-platform/riscv-isa-sim) for both IIS machines and the Ubuntu-based Docker container.

Next steps involve:
- [ ] Install `spike-dasm` on the Docker container from the official release
- [ ] Install `spike-dasm` on the Gitlab CI and add installation instructions to the getting started guide

**Pull request merged**